### PR TITLE
feat(Tabs): add vertical props to tabs

### PR DESCRIPTION
**Pull request by jonathan-herbault**
## DESCRIPTION

In this Pr, we add a new props `vertical` to display the tabs list vertically.
Also provide correct support with arrow key.

## HOW TO TEST

Show exemple `vertical`

## SCREENSHOTS / SCREEN RECORDINGS

image

## COMPATIBILITY

- [ ] Tested on Safari (desktop)
- [x] Tested on Chrome (desktop)
- [ ] Tested on Firefox (desktop)
- [ ] Tested on mobile device sizes
- [ ] Tested on tablet device sizes
- [ ] Tested on IOS Safari (either device or simulator)

## QA

- [ ] Thoroughly tested in local environment
- [ ] Added tests for all new features
- [ ] Added tests that considered edge cases

## Summary by CodeRabbit

* **New Features**
  * Added vertical orientation support for tabs, including full-width tabs and vertically stacked tab lists.
  * Added a vertical Tabs example with badges, an icon, and shared tab state.

* **Documentation**
  * Updated guidance and interactive examples for configuring tab orientation.
  * Refreshed Tabs property references and documentation links.

* **Tests**
  * Added coverage for default horizontal and configured vertical ARIA orientations, including vertical styling.

### CODE REVIEW

File: lib/src/components/Tabs/tests/index.test.tsx [hunk 1]
```diff
@@ -176,4 +176,36 @@ describe('Tabs', () => {
...
+        <Tabs aria-label="Tabs" store={tab}>
+          <Tabs.Tab data-testid="tab1" id="tab1" store={tab}>
+            Tab 1
+          </Tabs.Tab>
+          <Tabs.Tab data-testid="tab2" id="tab2" store={tab}>
+            Tab 2
+          </Tabs.Tab>
+        </Tabs>
+      )
+    }
```
Comment (cnairi) [inline lib/src/components/Tabs/tests/index.test.tsx]: You can change the TabsList component above for the following:
```
const TabsList = ({ options }: { options?: TabStoreProps }) => {
  const tab = useTab({ defaultSelectedId: 'tab1', ...options })

  return (
    <>
      <Tabs aria-label="Tabs" store={tab}>
        <Tabs.Tab data-testid="tab1" id="tab1" store={tab}>
          Tab 1
        </Tabs.Tab>
        <Tabs.Tab data-testid="tab2" id="tab2" store={tab}>
          Tab 2
        </Tabs.Tab>
        <Tabs.Tab data-testid="tab3" disabled id="tab3" store={tab}>
          Tab 3
        </Tabs.Tab>
      </Tabs>
    </>
  )
}
```

and then, replace your `  render(<TabsListVertical />)`  with `render(<TabsList options={{ orientation: 'vertical' }} />)`

File: lib/src/components/Tabs/index.tsx [hunk 2]
```diff
@@ -15,14 +15,17 @@ const cx = classNames(styles)
 
 export const TabsComponent = forwardRef<HTMLDivElement, TabListProps>(
   ({ children, className, size = 'lg', store, ...rest }, ref) => {
+    const { orientation } = useStoreState(store)
+    const vertical = orientation === 'vertical'
```
Comment (cnairi) [inline lib/src/components/Tabs/index.tsx]: Not sure about the naming here, I would go with `isVertical` that fits more our usual conventions please

File: lib/src/components/Tabs/Tab.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import { Badge } from '@/components/Badge'
 import { classNames } from '@/utils'
 import { forwardRefWithAs } from '@/utils/forwardRefWithAs'
 
-import { useTabSize } from './TabContext'
+import { useTabSize, useTabVertical } from './TabContext'
```
Comment (cnairi) [inline lib/src/components/Tabs/Tab.tsx]: Hum since we already have a `useTabSize`, I think a more persistent naming would be `useTabOrientation`.

I would have a default value = horizontal

And then you'll change it to vertical according to the tabs orientation

File: lib/src/components/Tabs/index.tsx [hunk 4]
```diff
@@ -15,14 +15,17 @@ const cx = classNames(styles)
...
     return (
       <AriakitTabList
-        className={cx('tab-list', `size-${size}`, className)}
+        className={cx('tab-list', `size-${size}`, vertical && 'orientation-vertical', className)}
         ref={ref}
         store={store}
         {...rest}
       >
-        <TabContext.Provider value={{ size }}>{children}</TabContext.Provider>
+        <TabContext.Provider value={{ size, vertical }}>{children}</TabContext.Provider>
```
Comment (cnairi) [inline lib/src/components/Tabs/index.tsx]: I think, since we have `size`, that it should be `orientation` instead of `vertical` cf https://github.com/WTTJ/welcome-ui/pull/3349/changes#r4080663254

**Review by cnairi — APPROVED** [review]
